io.go

```diff
@@ -6,16 +6,38 @@ import (
 	"io"
 )
 
-func Copy(ch chan<- []byte, r io.Reader) error {
+func Copy(ch chan<- []byte, r io.Reader, w io.Writer) error {
 	scanner := bufio.NewScanner(r)
 
 	for scanner.Scan() {
-		ch <- []byte(scanner.Text() + "\n")
+		line := scanner.Bytes()
+		cpy := make([]byte, len(line)+1)
+		copy(cpy, line)
+		line = append(line, '\n')
+
+		if err := writeFully(w, line); err != nil {
+			return err
+		}
+
+		ch <- []byte(line)
 	}
 
 	return scanner.Err()
 }
 
+func writeFully(w io.Writer, bytes []byte) error {
```

Comment:
I'm a bit surprised there isn't any `ioutil.WriteAll` or similar, as there is a `ReadAll`.